pkg/dns/azure/client/auth.go

```diff
@@ -1,21 +1,61 @@
 package client
 
 import (
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/jongio/azidext/go/azidext"
 )
 
 func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
-	oauthConfig, err := adal.NewOAuthConfig(
-		config.Environment.ActiveDirectoryEndpoint, config.TenantID)
+	var cloudConfig cloud.Configuration
+	switch config.Environment {
+	case azure.ChinaCloud:
+		cloudConfig = cloud.AzureChina
+	// GermanCloud was closed on Oct 29, 2021
```

Comment:
Do we have any CI jobs that still expect the GermanCloud region/zone to be present?

Reply:
I didn't find anything in openshift/release.

Reply:
But I can un-comment it if you think it's safer.